scripts/tests/linux/log_line_processing.py

```diff
@@ -63,26 +63,30 @@ def _io_thread(self):
         err_wait = select.poll()
         err_wait.register(self.process.stderr, select.POLLIN | select.POLLHUP)
 
-        with open(self.output_path, "wt") as f:
+        with open(self.output_path, "wt", buffering=1) as f:
             f.write("PROCESS START: %s\n" % time.ctime())
+            f.flush()
             while not self.done:
-                changes = out_wait.poll(0.1)
-                if changes:
-                    out_line = self.process.stdout.readline()
-                    if not out_line:
-                        # stdout closed (otherwise readline should have at least \n)
-                        continue
-                    f.write(out_line)
-                    self.output_lines.put(out_line)
-
-                changes = err_wait.poll(0)
-                if changes:
-                    err_line = self.process.stderr.readline()
-                    if not err_line:
-                        # stderr closed (otherwise readline should have at least \n)
-                        continue
-                    f.write(f"!!STDERR!! : {err_line}")
+                out_changes = out_wait.poll(0.1)
+                for fd, _ in out_changes:
+                    if fd == self.process.stdout.fileno():
+                        while True:
```

Comment:
The `while True` loop allows us to read all available data from the output stream and store it in the output queue. This change seems to alleviate the race condition we had with the test script. If I remember correctly from debugging, previously we were adding one line to the output queue at a time when data was available. However, if the output queue was empty while trying to retrieve and process data from it, we seemed to repeatedly enter the `_io_thread`, poll for output, and then timing out without reaching `self.process.stdout.readline()`. 

When running the tv-app and the tv-casting-app manually in separate terminals, I've observed that we receive a lot of messages at once with intermittent pauses (of maybe a few milliseconds) before the next big stream of messages appear. This suggests, as you pointed out, that we likely don't block for long when doing `self.process.stdout.readline()`.

I just did a quick check and it looks like once we enter the `while True`, we do exit the loop and enter it again.

I've also re-ran the [CI test 20 times in my local branch](https://github.com/shaoltan-amazon/connectedhomeip/actions/runs/11583471460) and this change passes for all 20 attempts.

Reply:
Unresolving: I have tested your PR and we only re-enter the loop after the file closes (i.e. we seem to always read all the items, then we repeateadly read 0 a couple of times). The logs I have look like:

- `casting app` reads 1507 lines
- `casting app` reads 0 lines
- `casting app` reads 0 lines
- ...
- `casting app` reads 0 lines
- `tv app` reads 5585 lines

This effectively makes this entire read single-threaded. This would be ok if we would not split out stdout and stderr. Maybe if we really need this we should split int two I/O threads that are doing blocking readson on each of stdout and stderr.

Reply:
Discussed offline. Made the changes to now create two threads (stdout and stderr) and read all lines.